### PR TITLE
New post about XTDB and Phoenix

### PR DESCRIPTION
**Pull request by tatut**
New post about using XTDB with Erlang ecosystem Phoenix Framework specifically.

### CODE REVIEW

File: _posts/2023-01-03-xtdb-phoenix.md [hunk 1]
```diff
@@ -0,0 +1,304 @@
...
+but there are other interesting options to consider. There has been a lot of interest in Clojure circles
+related to Datalog databases, like [Datomic](https://www.datomic.com)
+and [XTDB](https://xtdb.com), which provide a different model of programming especially as it relates
+to time.
+
+Datalog need not only be for Clojure programmers, many programming languages can use it, especially ones
+that have good facilities for describing data.
+In this post, I will give a short introduction to a new library that allows using XTDB with the Erlang
+ecosystem (specifically Elixir) and how to integrate that with the popular [Phoenix Framework](https://phoenixframework.org).
+We will briefly introduce Datalog and the library, then create a sample app with a simple
```
Comment (hjhamala) [inline _posts/2023-01-03-xtdb-phoenix.md:24]: Keep the pronomine same ->  I will give ... I will briefly

File: _posts/2023-01-03-xtdb-phoenix.md [hunk 2]
```diff
@@ -0,0 +1,304 @@
...
+LiveView component.
+
+## Datalog, what?
+
+What is datalog then and why should a web developer care about it? Datalog provides a different
```
Comment (hjhamala) [inline _posts/2023-01-03-xtdb-phoenix.md:29]: "What is datalog then and why should a web developer care about it?" replicates unneededly the header that says the same thing.

File: _posts/2023-01-03-xtdb-phoenix.md [hunk 3]
```diff
@@ -0,0 +1,304 @@
...
+programming model from the relational "table" view of things and instead focuses on entities that
+have attributes. Entities are open in that they can have any attributes without limitations, so
+there is no need to know all the columns beforehand or write migrations to add them.
+
+![SQL table vs Datalog facts](/img/2023-xtdb-phoenix/table-vs-facts.png)
+
+Instead of tables having columns, you have entity attribute value triples (EAV) like
+`[:bob :likes :pizza]` and `[:bob :date-of-birth 1997-04-09]`. Even joins are similarly
+marked by having an attribute value be the identity of another entity.
```
Comment (hjhamala) [inline _posts/2023-01-03-xtdb-phoenix.md:38]: Does xdtb have referential integrity enforced if value is pointer to a different entity?

File: _posts/2023-01-03-xtdb-phoenix.md [hunk 4]
```diff
@@ -0,0 +1,304 @@
...
+Relational SQL databases are still the "de facto" choice for most new web applications these days,
+but there are other interesting options to consider. There has been a lot of interest in Clojure circles
+related to Datalog databases, like [Datomic](https://www.datomic.com)
+and [XTDB](https://xtdb.com), which provide a different model of programming especially as it relates
+to time.
+
+Datalog need not only be for Clojure programmers, many programming languages can use it, especially ones
+that have good facilities for describing data.
+In this post, I will give a short introduction to a new library that allows using XTDB with the Erlang
+ecosystem (specifically Elixir) and how to integrate that with the popular [Phoenix Framework](https://phoenixframework.org).
```
Comment (hjhamala) [inline _posts/2023-01-03-xtdb-phoenix.md:23]: Consider shortly introducing what is Elixir in the Erlang ecosystem.

File: _posts/2023-01-03-xtdb-phoenix.md [hunk 5]
```diff
@@ -0,0 +1,304 @@
...
+  end
+
+  def search(assigns) do
+    ~H"""
+    <input type="text" phx-keyup="update_search" phx-debounce="300" placeholder="Search by email"/>
+    """
+  end
+
+  def results(assigns) do
+    case assigns.results do
```
Comment (hjhamala) [inline _posts/2023-01-03-xtdb-phoenix.md:229]: Perhaps comment the pattern matching to happen for the people who dont read easily Elixir code.

File: _posts/2023-01-03-xtdb-phoenix.md [hunk 6]
```diff
@@ -0,0 +1,304 @@
...
+```
+
+Then we run `mix deps.get` to fetch all the dependencies and we are ready to launch!
+
+Launch an interactive shell and the application by using the command:
+```shell
+iex --erl "-sname xthello@localhost" -S mix phx.server
+```
+
+You should see startup messages and a URL that points you to `http://localhost:4000`.
```
Comment (hjhamala) [inline _posts/2023-01-03-xtdb-phoenix.md:155]: What kind of files the initalization did create or what kind of directory structure is present?

I found little bit hard to follow what were autogenerated for the application and what was added.

**Review by hjhamala — APPROVED** [review]
Quite good in this way but check the comments whether you will find those valuable.

File: _posts/2023-01-03-xtdb-phoenix.md (hunk 3 above)
Comment (tatut) [reply _posts/2023-01-03-xtdb-phoenix.md:38]: They are just values, but you can do such checks with transaction functions. Don't know if I should mention or if the link to XTDB learning course is enough.

**Review by philosotractor — APPROVED** [review]
LGTM. 

Good that there's code comments int the snippets to explain main points to reader not familiar with the language.
